src/char/char_clif.c

```diff
@@ -523,6 +527,36 @@ int chclif_parse_char_delete2_req(int fd, struct char_session_data* sd) {
 	return 1;
 }
 
+/**
+ * Check char deletion code
+ * @param sd
+ * @param delcode E-mail or birthdate
+ * @param flag Delete flag
+ * @return 1:Success, 0:Failure
+ **/
+static int chclif_delchar_check(struct char_session_data *sd, char *delcode, uint8 flag) {
+	// E-Mail check
+	if (flag&1) {
+		if (strcmp(delcode, sd->email) && //email does not matches and
+			(strcmp("a@a.com", sd->email) || //it is not default email, or
+			(strcmp("a@a.com", delcode) && strcmp("", delcode)) //email sent does not matches default
+			))
```

Comment:
I know this check is not coming from you, but it would also be nice to have that one changed check for a match and not a "not match" and having a no operation inside the if block. Also same as below the two ifs should be merged.
I would suggest that way:
```
!strcmp(delcode, sd->email) || //email does match or
(
	!strcmp("a@a.com", sd->email) && //it is default email and
	(
		!strcmp("a@a.com", delcode) || // user also sent the default email 
		!strcmp("", delcode)) // user sent an empty email
	)
)
```